src/client.ts

```diff
@@ -1638,26 +1640,37 @@ export class StreamChat<StreamChatGenerics extends ExtendableGenerics = DefaultG
   hydrateActiveChannels(
     channelsFromApi: ChannelAPIResponse<StreamChatGenerics>[] = [],
     stateOptions: ChannelStateOptions = {},
+    queryChannelsOptions?: ChannelOptions,
   ) {
     const { skipInitialization, offlineMode = false } = stateOptions;
-
-    for (const channelState of channelsFromApi) {
-      this._addChannelConfig(channelState.channel);
-    }
-
     const channels: Channel<StreamChatGenerics>[] = [];
 
     for (const channelState of channelsFromApi) {
+      this._addChannelConfig(channelState.channel);
       const c = this.channel(channelState.channel.type, channelState.channel.id);
       c.data = channelState.channel;
       c.offlineMode = offlineMode;
       c.initialized = !offlineMode;
 
+      let updatedMessagesSet;
       if (skipInitialization === undefined) {
-        c._initializeState(channelState, 'latest');
+        const { messageSet } = c._initializeState(channelState, 'latest');
+        updatedMessagesSet = messageSet;
       } else if (!skipInitialization.includes(channelState.channel.id)) {
         c.state.clearMessages();
-        c._initializeState(channelState, 'latest');
+        const { messageSet } = c._initializeState(channelState, 'latest');
+        updatedMessagesSet = messageSet;
+      }
+
+      if (updatedMessagesSet) {
+        updatedMessagesSet.pagination = {
```

Comment:
Pagination indicators are calculated once the requested page is merged into the parent message set.